### ISSUE TITLE
Force Unwrapping false positive

### ISSUE THREAD

**Issue by AliBilal** [realm/SwiftLint#977]
Declaring a local variable inside a function
`var a: [Int]!`
is considered force_unwrapping violation. But we are just defining an implicitly unwrapped optional and not force unwrapping it. Seems it is a false positive. It only occurs for Array type and not plain types. e.g. `var a: Int!` is not considered rule violation.

**Comment by JamieEdge** [comment]
Which version are you using?

I cannot reproduce this with the following example:

![Example](https://cloud.githubusercontent.com/assets/1623997/21456779/75fd39c0-c922-11e6-86fb-d58be2e69e74.png)

**Comment by AliBilal** [comment]
I'm currently using version 0.13.2

Please make sure to define the variable inside a function or method. For example:

`func foo() {
    var x: [Int]!
}`

The `force_unwrapping` rule is an opt-in rule. So please also make sure it is opted in.

**Comment by marcelofabri** [comment]
I was able to reproduce it on `master`.

**Comment by JamieEdge** [comment]
Sorry, moved my test files but not the config! Can now reproduce the issue.

**Comment by marcelofabri** [comment]
I think this is actually a consequence of https://github.com/realm/SwiftLint/issues/136.

```bash
$ sourcekitten structure --text "func foo() { var x: [Int]! }"
{
  "key.substructure" : [
    {
      "key.kind" : "source.lang.swift.decl.function.free",
      "key.offset" : 0,
      "key.nameoffset" : 5,
      "key.namelength" : 5,
      "key.bodyoffset" : 12,
      "key.bodylength" : 15,
      "key.accessibility" : "source.lang.swift.accessibility.internal",
      "key.length" : 28,
      "key.name" : "foo()"
    }
  ],
  "key.offset" : 0,
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 28
}
```

**Comment by marcelofabri** [comment]
Fixed in #1620.
